ner_v2/detectors/temporal/date/hi/date_detection.py

```diff
@@ -0,0 +1,59 @@
+from chatbot_ner.config import ner_logger
+from ner_v2.detectors.temporal.constant import LANGUAGE_DATA_DIRECTORY
+from ner_v2.detectors.temporal.date.standard_regex_date import BaseRegexDate
+import datetime
+import pytz
+import os
+
+
+class DateDetector(BaseRegexDate):
+    def __init__(self, entity_name, timezone='UTC'):
+
+        self.text = ''
+        self.tagged_text = ''
+        self.processed_text = ''
+        self.date = []
+        self.original_date_text = []
+        self.entity_name = entity_name
+        self.tag = '__' + entity_name + '__'
+        try:
+            self.timezone = pytz.timezone(timezone)
+        except Exception as e:
+            ner_logger.debug('Timezone error: %s ' % e)
+            self.timezone = pytz.timezone('UTC')
+            ner_logger.debug('Default timezone passed as "UTC"')
+        self.now_date = datetime.datetime.now(tz=self.timezone)
+        self.bot_message = None
+
+        data_directory_path = os.path.join(os.path.dirname(os.path.abspath(__file__)).rstrip(os.sep),
+                                           LANGUAGE_DATA_DIRECTORY)
+
+        super(DateDetector, self).__init__(data_directory_path=data_directory_path)
+
+    def detect_date(self, text):
```

Comment:
How are you ensuring that every individual detector has detect_date function. Make it an abstractmethod in BaseRegexDate itself. Refer to base detector for more clarity.